circuits/cpp/src/aztec3/circuits/hash.hpp

```diff
@@ -85,6 +85,32 @@ template <typename NCT> typename NCT::address compute_contract_address(Point<NCT
     return compute_contract_address_from_partial(point, partial_address);
 }
 
+template <typename NCT>
+typename NCT::fr compute_commitment_nonce(typename NCT::fr first_nullifier, typename NCT::fr commitment_index)
+{
+    using fr = typename NCT::fr;
+
+    std::vector<fr> const inputs = {
+        first_nullifier,
+        commitment_index,
+    };
+
+    return NCT::hash(inputs);
```

Comment:
No domain separators? :)

Reply:
Not yet. But will definitely add that :) Trying to get the flow setup correctly first.